### ISSUE TITLE
Missing support for nonverbal AT responses, such as earcons and haptics, impacts report accuracy

### ISSUE THREAD

**Issue by mcking65** [w3c/aria-at-app#1661]
There are two significant problems caused by the lack of support for nonverbal AT responses in ARIA-AT:
1. Some bot-generated reports may contain inaccuracies.
2. Accurate reports can appear inaccurate due to missing nonverbal response data (and vice versa).

#### Essential Context

In their default configuration, AT may respond to a command with output that is nonverbal. For instance, they may use audible earcons or haptics to convey semantics or a change in the value of an AT setting. Two examples are:
1. VoiceOver on macOS conveys the alert role via an alert sound. It does not include a string in its speech output that represents the alert role.
2. JAWS and NVDA convey the automatic change in their active cursor setting (e.g., JAWS virtual cursor to PC cursor) via a system sound.

Note:
* VoiceOver does not have a setting that enables replacement of the alert role earcon with a speech output string.
* VoiceOver includes a braille output representation of the alert role.
* JAWS and NVDA have settings for replacing the active cursor change sound with a spoken string.

#### Problem 1: Some Bot-Generated Reports May Contain Inaccuracies

Because current AT Driver implementations do not include features for detecting and capturing nonverbal responses, such as system sounds or haptics, earcons are a form of AT response that cannot be detected by the bots. This could impact the accuracy of bot-generated reports containing assertion verdicts that depend on observation of a nonverbal AT response.

When a bot is generating a report, after it captures the AT response to a command, it assigns assertion verdicts by comparing the response it collected to responses previously analyzed by humans. Because the collected responses do not include any information about the nonverbal aspect of the response, the bot will perpetuate the previous verdicts, regardless of whether the nonverbal behavior of the AT is the same.

For example, consider a scenario where the VO bot is generating a report for a new version of VoiceOver (version N+1). The system contains a report for VoiceOver version N. In the report for version N, a human recorded a verdict of "pass" for the assertion that VoiceOver conveyed the alert role because the human heard the alert earcon. Now the bot runs the test plan with VoiceOver version N+1 and collects the same verbal response that was recorded for VoiceOver version N. VO bot will assign the verdict of "pass" to the assertion about the alert role because the two responses match. However, if VoiceOver version N+1 did not generate the alert earcon, the bot-generated report would be inaccurate and the current system would not raise any flags; the report would be automatically published if all version N+1 verbal responses match all version N verbal responses.

The current system does not identify which assertion verdicts depend on nonverbal responses. If it did, it could avoid automatically assigning verdicts without human intervention.

In the long term, it would be ideal if the AT Driver API could support nonverbal responses. In the meantime, there will be verdicts that have this constraint.

#### Problem 2: Accurate Reports Can Appear Inaccurate Due to Missing Nonverbal Response Data

When recording the AT response to a command, the manual test runner enables testers to capture the text representing speech output. That text string is included in reports, enabling report consumers to understand the verdicts that were assigned for each assertion. If the command does not generate speech output, there is a checkbox that enables the tester to positively record that observation. When that box is checked, reports will include a statement explaining that output was not observed.

However, when an AT response includes relevant output that is not speech, that output cannot be recorded in the manual test runner and such output cannot be described in reports. This could make an accurate report appear inaccurate.

For example, consider the VoiceOver report for conveying the alert role. VoiceOver conveys the role via an earcon. The speech output recorded for the commands in the report for the alert role do not indicate that the alert role was conveyed. However, because human testers have observed the earcon, as a result of feedback from Apple and the discussion in w3c/aria-at#1316, the community group has agreed to record a verdict of "pass" for this assertion. This leads to a report that may appear inaccurate because the response recorded in the report would not include evidence that supports the passing verdict.

Note that this problem also exists for all assertions that JAWS or NVDA changed the active cursor in response to the `Tab` command. The AT response included in reports does not include any evidence that the assertion passed or failed. The community group has previously discussed whether or not this justifies a change to default settings and concluded that testers should continue to use default settings but still record the appropriate verdict. [citation needed]
